src/ai/backend/manager/models/vfolder.py

```diff
@@ -108,6 +109,19 @@ class VFolderInvitationState(str, enum.Enum):
     REJECTED = "rejected"  # rejected by invitee
 
 
+class VFolderOperationStatus(str, enum.Enum):
+    """
+    Introduce virtual folder status for storage-proxy operations.
+    """
+
+    READY = "ready"
+    PERFORMING = "performing"
+    CLONING = "cloning"
+    DELETING = "deleting"
+    DELETED = "deleted"
+    MOUNTED = "mounted"
```

Comment:
Your implementation accepts desired vFolder status as an enum, which is also used to represent vFolder current status. The problem is that desired vFolder status and current vFolder status slightly differs its meaning. "current" vFolder status should represent as-is, e.g. Copying something, Cloning to new vFolder, being deleted, and so on. In contrast to previous term, "desired" condition should only distinguish between "ready to read" (superset of all created vFolders) or "ready to alter" (this should mean all vFolders which are only in "READY" status). Using same enum to represent two different categories can be confused to other readers reading this code. How about creating new Enum to represent desired vFolder condition? This value should be determined by current vFolder status (`vfolder.status` column), so it won't be added to database scheme.

```suggestion
class VFolderOperationStatus(str, enum.Enum):
    """
    Introduce virtual folder status for storage-proxy operations.
    """

    READY = "ready"
    PERFORMING = "performing"
    CLONING = "cloning"
    DELETING = "deleting"
    DELETED = "deleted"
    MOUNTED = "mounted"

class VFolderAccessStatus(str, enum.Enum):
    READABLE = "readable"
    WRITABLE = "writable"
```